azure/timer.ts

```diff
@@ -12,39 +12,62 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import * as azure from "@pulumi/azure";
+import * as subscription from "@pulumi/azure-serverless/subscription";
 import { timer } from "@pulumi/cloud";
 import * as pulumi from "@pulumi/pulumi";
 import { RunError } from "@pulumi/pulumi/errors";
-// import { Function } from "./function";
+import * as shared from "./shared";
 
 export function interval(name: string, options: timer.IntervalRate, handler: timer.Action,
                          opts?: pulumi.ResourceOptions): void {
-    let rateMinutes = 0;
-    if (options.minutes) {
-        rateMinutes += options.minutes;
+
+    if (options.days !== undefined) {
+        if (options.days !== 0) {
+            // Yes.  This is strange.  But it is how Azure interprets things:
+            // https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer#timespan
+            checkRange("options.days", options.days, 24, 100);
```

Comment:
I'm a little confused by the checks here.  Are there values that AWS can accept that Azure can't?  Or do they just need to be formatted differently to be passed to Azure? If the latter, why not do the reformatting for the user so that the accepted set of values is consistent for the `@pulumi/cloud` API?

That is - why not do something a bit more like the AWS implementation where we turn the minutes, hours, days into a total minutes and then pass (or reformat into a valid format) that value?

Reply:
Also - feels like we're diverging a fair bit between AWS and Azure implementations in how strict we are on handling of inputs here - we aggressively throw in Azure, but are fairly loose in AWS (things coerce down to integer minutes instead of throwing - so "`1.5 days" will work).  Feels like we want to be more consistent here.

Reply:
I've added this to https://github.com/pulumi/pulumi-cloud/issues/565 to track.